app/models/list_import/anilist/row.rb

```diff
@@ -1,98 +1,126 @@
+# frozen_string_literal: true
+
 class ListImport
   class Anilist
     class Row
-      attr_reader :node, :type
-
       def initialize(node, type)
         @node = node
-        @type = type # anime || manga
+        @type = type.to_s # anime | manga
       end
 
-      def media
-        key = "#{type}/#{media_info[:id]}"
+      def media_mapping
+        anilist_mapping = Mapping.lookup(anilist_key, media.id)
+
+        return anilist_mapping if anilist_mapping.present?
+
+        other_mapping = Mapping.lookup(mal_key, mal_id) ||
+                        Mapping.guess(type.classify.safe_constantize, media_info)
+
+        Mapping.create(item: other_mapping, external_site: anilist_key, external_id: media.id) if other_mapping.present?
+        other_mapping
+      end
+
+      def data
+        fields.map { |field| [field, send(field)] }.to_h.compact
+      end
+
+      private
+
+      attr_reader :node, :type
 
-        Mapping.lookup('anilist', key) ||
-          Mapping.guess(type.classify.safe_constantize, media_info)
+      def fields
+        %i[
+          rating status reconsume_count progress
+          notes started_at finished_at
+        ]
       end
 
+      # For mapping guess
       def media_info
         {
-          id: node[type]['id'],
-          title: node[type]['title_romaji'],
-          subtype: node[type]['type'],
-          episode_count: node[type]['total_episodes'],
-          chapter_count: node[type]['total_chapters']
+          title: title,
+          subtype: type,
+          episode_count: media.episodes,
+          chapter_count: media.chapters
         }.compact
       end
 
+      # 100-point scale to 20-point scale (raw)
+      # rating -> score
+      def rating
+        return nil if node.score.zero?
+
+        [(node.score.to_f / 5).ceil, 2].max
+      end
+
       def status
-        case node['list_status']
+        case node.status.downcase
         when 'completed' then :completed
-        when /ing\z/ then :current
-        when /\Aplan/ then :planned
-        when 'on-hold' then :on_hold
+        when 'current' then :current
+        when 'planning' then :planned
+        when 'paused' then :on_hold
         when 'dropped' then :dropped
         end
       end
 
-      def progress
-        type == 'anime' ? node['episodes_watched'] : node['chapters_read']
+      # reconsume_count -> repeat
+      def reconsume_count
+        node.repeat
       end
 
-      def volumes
```

Comment:
Shouldn't we still import this value? should be able to use `progress_volumes` right?

Reply:
good catch! I totally forgot to readd this.

Reply:
I am not seeing a `progress_volume` anywhere. All I can see in the schema is `volumes_owned` which is updated from library entry (Volume)..... Well that makes no sense but I can shove the amount in there? 
<img width="450" alt="Screen Shot 2020-04-24 at 5 32 15 PM" src="https://user-images.githubusercontent.com/1283983/80266614-9f38e380-8651-11ea-8b5e-c8efdc088b76.png">

Reply:
`progress_volumes` is from anilist, and it should map to `volumes_owned` (misnomer) on our end